docroot/modules/custom/foia_file/drush/foia_file.drush.inc

```diff
@@ -0,0 +1,198 @@
+  $config = \Drupal::config('file_entity.settings');
+  $extensions = $config->get('default_allowed_extensions');
+  $pipeDelimitedExtensions = str_replace(" ", "|", $extensions);
+  $scans = explode("\n", $scanOutput);
+
+  foreach ($scans as $scan) {
+    if (!isset($scan) || is_null($scan)) {
+      $msg = 'A blank line was output by the virus scanner.';
+      $msg .= 'This is not expected behavior.';
+      \Drupal::logger('foia_file')->warning($msg);
+    }
+
+    // Split the scan line into filename and status.
+    preg_match('/[^ ]*$/', $scan, $wordsArray);
+    // Set status to the last word of the line.
+    $status = $wordsArray[0];
+
+    $validStates = ['OK', 'Removed.', 'FOUND'];
+    if (!in_array($status, $validStates)) {
+      $msg = 'An invalid status was detected in the virus scan output.';
+      $msg .= 'Virus scanning will not continue until this is resolved!!!';
+      \Drupal::logger('foia_file')->error($msg);
+      return drush_set_error('POLICY_PM_DENY', dt($msg));
+    }
+
+    // Get relative file path.
+    $scanLength = strlen($scan);
+    $statusLength = strlen($status);
+    $fileLength = $scanLength - $statusLength - 2;
+    $fileNameWithVirusName = substr($scan, 0, $fileLength);
+
+    // Trim virus description.
+    preg_match('/(\.(' . $pipeDelimitedExtensions . '))(: .*)?$/', $fileNameWithVirusName, $fileExtension);
+    $delimiter = $fileExtension[1];
+    $fileName = substr($scan, 0, strpos($fileNameWithVirusName, $delimiter));
+    $fileNameWithExtension = $fileName . $fileExtension[1];
+
+    if ($status === 'OK' && !is_file($fileNameWithExtension)) {
+      $msg = "The file {$fileNameWithExtension} was not detected on the filesystem and was marked as clean by the virus scanner.";
+      \Drupal::logger('foia_file')->warning($msg);
+    }
+
+    if ($status === 'FOUND' || $status === 'Removed.') {
+      $msg = "The file {$fileNameWithExtension} was detected on the filesystem and was marked as infected by a virus. It should have been deleted by the scanner.";
+      \Drupal::logger('foia_file')->warning($msg);
+    }
+  }
+
+  $endTime = microtime(TRUE);
+
+  $executionTime = ($endTime - $startTime);
+
+  \Drupal::logger('foia_file')->info("File Entity update validation completed in {$executionTime} seconds.");
+
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_foia_file_file_entity_update($scanOutput) {
+  $startTime = microtime(TRUE);
+
+  // Get webform directory on the server.
+  $fileDir = explode('/webform/', $scanOutput, 2);
+  $webformDir = "{$fileDir[0]}/webform/";
+
+  // Temporarily delete similar paths of the webform dir during processing.
+  $trimmedFiles = str_replace($webformDir, '', $scanOutput);
+
+  // Split string into lines. @todo change to preg split \R?
+  $scans = explode("\n", $trimmedFiles);
+
+  // Get allowed extensions.
+  $config = \Drupal::config('file_entity.settings');
+  $extensions = $config->get('default_allowed_extensions');
```

Comment:
Let's talk this through. What happens if this code is passed a file that is not one of the above extensions? For example, as of right now webform submissions can contain `odf` files, but that isn't one of the extensions that will get returned here. Do we know how the code will handle that scenario? I wonder if this code needs to be more generic to be able to account for any file extension